code/game/turfs/walls/wall_types.dm

```diff
@@ -14,6 +14,23 @@
 	density = TRUE
 
 	base_icon_state = "testwall"
+	///do we have bits of decoration to add to the walls?
+	var/decorated_wall = TRUE
+
+/turf/closed/wall/mainship/smooth_icon()
+	. = ..()
+	cut_overlays()
+	if(decorated_wall)
+		var/r1 = rand(0,10) //Make a random chance for this to happen
+		var/r2 = rand(0,3) // Which wall if we do choose it
+		if(length(canSmoothWith) && !CHECK_MULTIPLE_BITFIELDS(smoothing_junction, (WEST_JUNCTION)))
+			return
+		if(CHECK_MULTIPLE_BITFIELDS(smoothing_junction, (SOUTH_JUNCTION)) || !CHECK_MULTIPLE_BITFIELDS(smoothing_junction, (EAST_JUNCTION)))
+			return
+		else
+			switch(r1)
+				if(9 to 10)
+					add_overlay("[r2]")
```

Comment:
what is this logic? do you mean
```suggestion
		if(r1 == 9 || r1 == 10)
			add_overlay("[r2]")
```

Reply:
> what is this logic? do you mean

It's the original logic used for the wall decorations, slightly adjusted to use overlays instead of an icon_state